### PR TITLE
Add ruby support

### PR DESCRIPTION
**Pull request by tim-coutinho**
This addresses #38.

### CODE REVIEW

File: shallow_backup/reinstall.py [hunk 1]
```diff
@@ -165,7 +165,9 @@ def run_cmd_if_no_dry_run(command, dry_run) -> int:
 		elif pm == "macports":
 			print_red_bold("WARNING: Macports reinstallation is not supported.")
 		elif pm == "gem":
-			print_red_bold("WARNING: Gem reinstallation is not supported.")
+			print_pkg_mgr_reinstall(pm)
+			cmd = f"cat {packages_path}/gem_list.txt | xargs -L 1 sudo gem install --no-ri --no-rdoc"
```
Comment (alichtman) [inline shallow_backup/reinstall.py:169]:
```suggestion
			cmd = f"cat {packages_path}/gem_list.txt | xargs -L 1 gem install --no-ri --no-rdoc"
```

Comment (alichtman) [reply shallow_backup/reinstall.py:169]: We don't want to install gems with `sudo`. Huge security risk.

File: shallow_backup/reinstall.py [hunk 2]
```diff
@@ -165,7 +165,9 @@ def run_cmd_if_no_dry_run(command, dry_run) -> int:
 		elif pm == "macports":
 			print_red_bold("WARNING: Macports reinstallation is not supported.")
 		elif pm == "gem":
-			print_red_bold("WARNING: Gem reinstallation is not supported.")
+			print_pkg_mgr_reinstall(pm)
+			cmd = f"cat {packages_path}/gem_list.txt | xargs -L 1 gem install --no-ri --no-rdoc"
```
Comment (alichtman) [inline shallow_backup/reinstall.py:169]:
```suggestion
			cmd = f"cat {packages_path}/gem_list.txt | xargs -L 1 gem install"
```

Comment (alichtman) [reply shallow_backup/reinstall.py:169]: Let's let our users keep their documentation too :)

**Comment by alichtman** [comment]
The reinstallation fails on my system.

```bash
$ gem list | tail -n+1 | sed 's/(/--version /' | sed 's/)//' | xargs -L 1 gem install
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
Fetching bundler-unload-1.0.2.gem
Successfully installed bundler-unload-1.0.2
Parsing documentation for bundler-unload-1.0.2
Installing ri documentation for bundler-unload-1.0.2
Done installing documentation for bundler-unload after 0 seconds
1 gem installed
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]
ERROR:  While executing gem ... (Gem::Requirement::BadRequirementError)
    Illformed requirement ["default:"]

```

**Comment by tim-coutinho** [comment]
Interesting, it was working earlier when I was testing it on my machine, perhaps there was a recent change in `gem list`'s functionality that added the `default: ` to some version numbers.

File: shallow_backup/utils.py [hunk 3]
```diff
@@ -1,41 +1,47 @@
...
+		if process.returncode == 0:
+			with open(filepath, "w+") as f:
+				f.write(process.stdout.decode('utf-8'))
+		else:
+			print_path_red("An error occurred while running: $", command)
+
 		return process.returncode
 	else:
-		print_path_red("An error occurred while running: $", command)
+		print_yellow("Package manager not present.")
```
Comment (alichtman) [inline shallow_backup/utils.py:44]:
```suggestion
		print_path_red("An error occurred while running: $", command)
```

**Comment by alichtman** [comment]
Hmm, interesting. I'm using `gem 3.1.2`. We might need to run a different command to backup / reinstall with `gem`.

File: shallow_backup/utils.py (hunk 3 above)
Comment (alichtman) [reply shallow_backup/utils.py:44]: I think it makes sense to keep the error messages in `run_cmd_write_stdout` pretty abstract. It's a util function, and we let the calling function know of the error with the error code returned. A more specific error message can be printed from there.

Comment (tim-coutinho) [reply shallow_backup/utils.py:44]: True, though there won't be an error code to check in this case, as a `FileNotFoundError` results in a return of `None`. I can change it to something like `Command not found`, which is a bit more general and accurate to what a return of `None` represents.

**Comment by tim-coutinho** [comment]
My latest commit fixes the issue, just had to update the backup command.

**Comment by alichtman** [comment]
@tim-coutinho Sorry for taking so long to get around to this. Rebase off `master` and then I'll merge this.

File: shallow_backup/utils.py (hunk 3 above)
Comment (alichtman) [reply shallow_backup/utils.py:44]: Let's make this change and then I'll merge it.

**Pull request closed without merging**